From 225577f7a15b742ecc47297025b8a6caf621e8fe Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:47 -0400
Subject: [powerpc] Make unaligned accesses endian-safe for powerpc

Message-id: <1410545655-205645-239-git-send-email-dzickus@redhat.com>
Patchwork-id: 94258
O-Subject: [RHEL7 PATCH 238/626] powerpc: Make unaligned accesses endian-safe for powerpc
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit e8a00ad5e238421ded856ea39f692b94c2d324eb
Author: Rajesh B Prathipati <rprathip@linux.vnet.ibm.com>
Date:   Mon Dec 16 18:58:22 2013 +1100

    powerpc: Make unaligned accesses endian-safe for powerpc

    The generic put_unaligned/get_unaligned macros were made endian-safe by
    calling the appropriate endian dependent macros based on the endian type
    of the powerpc processor.

    Signed-off-by: Rajesh B Prathipati <rprathip@linux.vnet.ibm.com>
    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/unaligned.h b/arch/powerpc/include/asm/unaligned.h
index 5f1b1e3..8296381 100644
--- a/arch/powerpc/include/asm/unaligned.h
+++ b/arch/powerpc/include/asm/unaligned.h
@@ -4,13 +4,18 @@
 #ifdef __KERNEL__
 
 /*
- * The PowerPC can do unaligned accesses itself in big endian mode.
+ * The PowerPC can do unaligned accesses itself based on its endian mode.
  */
 #include <linux/unaligned/access_ok.h>
 #include <linux/unaligned/generic.h>
 
+#ifdef __LITTLE_ENDIAN__
+#define get_unaligned __get_unaligned_le
+#define put_unaligned __put_unaligned_le
+#else
 #define get_unaligned __get_unaligned_be
 #define put_unaligned __put_unaligned_be
+#endif
 
 #endif /* __KERNEL__ */
 #endif /* _ASM_POWERPC_UNALIGNED_H */
-- 
1.7.1